pyartcd/pyartcd/pipelines/doomsday.py

```diff
@@ -0,0 +1,74 @@
+import os
+from typing import Optional
+import click
+from time import sleep
+import shutil
+
+from pyartcd.runtime import Runtime
+from pyartcd.cli import cli, pass_runtime
+from artcommonlib.exectools import cmd_assert
+from doozerlib.util import mkdirs
+
+
+class DoomsdaySync:
+    ALL_ARCHES_LIST = ["x86_64", "s390x", "ppc64le", "aarch64", "multi"]
+    N_RETRIES = 4
+
+    def __init__(self, runtime: Runtime, all_arches: bool, arches: Optional[list[str]], version: str):
+        self.runtime = runtime
+        self.version = version
+        self.workdir = "./workspace"
+
+        if arches and all_arches:
+            raise Exception("Cannot specify both --arches and --all-arches")
+
+        if not (arches or all_arches):
+            raise Exception("Either --arches or --all-arches needs to be specified")
+
+        self.arches = arches.split(",") if not all_arches else self.ALL_ARCHES_LIST
+
+    def sync_arch(self, arch: str):
+        if arch not in self.ALL_ARCHES_LIST:
+            raise Exception(f"Invalid arch: {arch}")
+
+        major_minor = ".".join(self.version.split(".")[:2])
+        path = f"{self.workdir}/{major_minor}/{self.version}/{arch}"
+        mirror_cmd = f"oc adm release mirror quay.io/openshift-release-dev/ocp-release:{self.version}-{arch} --keep-manifest-list --to-dir={path}"
```

Comment:
Not part of your porting work but noticing there are more "multi" entries that might need backup? 

```
skopeo list-tags docker://quay.io/openshift-release-dev/ocp-release | grep '4\.16\.13'
        "4.16.13-aarch64",
        "4.16.13-multi",
        "4.16.13-multi-aarch64",
        "4.16.13-multi-ppc64le",
        "4.16.13-multi-s390x",
        "4.16.13-multi-x86_64",
        "4.16.13-ppc64le",
        "4.16.13-s390x",
        "4.16.13-x86_64",
```

Is `4.16.13-multi-x86_64` implicitly backed up when `4.16.13-multi` is treated?

Reply:
@adobes1 lets keep this as a follow up. I'll give you read only access to the AWS cluster, once this PR is merged, make sure that files are being uploaded correctly to the bucket, this week.